server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -111,7 +111,8 @@ static ByteSizeValue newByteSizeValue(long sizeInBytes, ByteSizeUnit desiredUnit
     public static ByteSizeValue readFrom(StreamInput in) throws IOException {
         long size = in.readZLong();
         ByteSizeUnit unit = ByteSizeUnit.readFrom(in);
-        if (in.getTransportVersion().onOrAfter(BYTE_SIZE_VALUE_ALWAYS_USES_BYTES)) {
+        // BYTE_SIZE_VALUE_ALWAYS_USES_BYTES was later reverted, so we use equals
```

Comment:
This should use `onOrAfter(825) && before(826)` to account for any extra patch version that are added between the two

Reply:
Or, better, `isPatchFrom(825)`